apps/src/code-studio/levels/freeResponse.js

```diff
@@ -0,0 +1,39 @@
+import { TestResults } from '../../constants';
+
+export default class FreeResponse {
+  constructor(levelId) {
+    this.levelId = levelId;
+
+    $(document).ready(function () {
+      var textarea = $(`textarea#level_${levelId}.response`);
+      textarea.blur(function () {
+        window.dashboard.codeStudioLevels.onAnswerChanged(levelId, true);
+      });
+      textarea.on('input', null, null, function () {
+        window.dashboard.codeStudioLevels.onAnswerChanged(levelId, false);
```

Comment:
Instead of calling `window.dashboard.codeStudioLevels.onAnswerChanged` I think we should be able to import this method from codeStudioLevels. This way our usage of the global will be restricted to haml and our entry page, and we can more easily remove it in the future.